include/pybind11/numpy.h

```diff
@@ -1865,9 +1865,13 @@ struct vectorize_helper {
         }
 
         auto result = returned_array::create(trivial, shape);
+#ifdef PYBIND11_DETECTED_CLANG_WITH_MISLEADING_CALL_STD_MOVE_EXPLICITLY_WARNING
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wreturn-std-move"
+#endif
 
         if (size == 0) {
-            return std::move(result);
+            return result;
         }
```

Comment:
If you remove the std::move are the pragmas necessary here, anymore?

Reply:
Here a GCC 11 & 12 warning is at conflict with a clang 7 through 12 warning.

When GCC 11 & 12 complained, I simply removed the `std::move`. (Thinking: "Let's see what happens.")
Next CI: many clang failures.
Next CI: complete the clang coverage to better understand what is going on.

Looking at the results I concluded (**guessed** TBH): clang was wrong for a long time, but that was fixed in clang 13. The fix appears to stick: clang 14 & 15 agree.

Next CI: with pragmas for clang added.

I considered introducing something like
```
PYBIND11_SUPPRESS_MISLEADING_CLANG_WARNING_CALL_STD_MOVE_EXPLICITLY_BEGIN
...
PYBIND11_SUPPRESS_MISLEADING_CLANG_WARNING_CALL_STD_MOVE_EXPLICITLY_END
```
or
```
    return PYBIND11_KNOWN_REDUNDANT_STD_MOVE(result);
```
but decided against it because there are only two affected locations, and the implementation you see now is easier to understand just looking at the local code, without having to look elsewhere.